File: src/main/java/szewek/flux/item/MetalItem.java
package szewek.flux.item;

import net.minecraft.item.Item;
import szewek.flux.util.metals.Metal;

public final class MetalItem extends Item {
	public Metal metal;

	public MetalItem(Properties properties, Metal metal) {
		super(properties);
		this.metal = metal;
	}
}

File: src/main/java/szewek/flux/item/MetalItem.kt
package szewek.flux.item

import net.minecraft.item.Item
import szewek.flux.util.Metal

class MetalItem(properties: Properties) : Item(properties) {
    var metal: Metal? = null
        private set

    fun withMetal(metal: Metal?): MetalItem {
        if (this.metal == null) this.metal = metal
        return this
    }

}